### PR TITLE
Add possibility to force a StackTrace capture when no frames found

### PR DESCRIPTION
**Pull request by jviolas**
When capturing an Exception in the same code that create it, no stacktrace is sent to Sentry.

See #260 for details.

This PR adds an option when an Exception is captured to force Raven to generate the stacktrace.

I've only added this option for `Exception` through `new SentryEvent(Exception exception, bool forceCaptureStackTrace = false)`. It could be added for messages but it would require to either change `SentryMessage` to have a possible StackTrace or consider those messages as `SentryException`

### CODE REVIEW

**Review by asbjornu — APPROVED** [review]

**Comment by jviolas** [comment]
@bruno-garcia I commited changes from your suggestion. '
Just dont really know why it's not building. It seems the tests I've added are failing but they work when I launch it locally. From the logs, it seems the "internal stack frame skipping" is not working when launched in the context of the build. Any idea why ?

File: src/app/SharpRaven/Data/SentryStacktrace.cs [hunk 1]
```diff
@@ -46,18 +46,26 @@ public class SentryStacktrace
...
             var trace = new StackTrace(exception, true);
+            
             var frames = trace.GetFrames();
 
-            if (frames == null)
+            if (frames == null && forceCaptureStackTrace)
+            {
+                // Skip first frame if called from this assembly (in case of force capture stacktrace)
+                frames = new StackTrace(true).GetFrames().SkipWhile(f => f.GetMethod().Module.Name == "SharpRaven.dll").ToArray();
+            }
```
Comment (bruno-garcia) [inline src/app/SharpRaven/Data/SentryStacktrace.cs]: One last thing: I suggest using `f.GetMethod()?.Module.Name == "SharpRaven"` here, unless I'm mistaken GetMethod can return null.

**Comment by bruno-garcia** [comment]
Honestly I'm not sure.
On Travis, it runs on Mono so there might be a surprise there.
You can check if it's running on Mono and skip your test. [Something like this](https://github.com/getsentry/dotnet-sentry-platform-abstractions/blob/dev/test/Sentry.PlatformAbstractions.Tests/RuntimeInfoTests.cs#L48). You can easily check with ` Type.GetType("Mono.Runtime", false) != null`

On Windows it should behave the same, under .NET Framework. Running the cake build on your machine works and on appveyor doesn't is very strange.

**Comment by jviolas** [comment]
It seems the StackTrace is empty on Mono. I'll try to debug this later.

**Comment by jviolas** [comment]
Yeah ! 🎉 Found it. So in Mono, `StackTrace.GetFrames()` returns an empty array instead of null when there is no frame. So my checks in `SentryStacktrace` constructor was not correct.

**Review by bruno-garcia — APPROVED** [review]
Looks good! Thanks

**Pull request merged**